Paintroid/src/main/java/org/catrobat/paintroid/presenter/MainActivityPresenter.java

```diff
@@ -664,14 +695,15 @@ public void onSaveImagePostExecute(@SaveImageRequestCode int requestCode, Uri ur
 			case SAVE_IMAGE_DEFAULT:
 				break;
 			case SAVE_IMAGE_FINISH:
-				navigator.finishActivity();
+				if (model.isOpenedFromCatroid()) {
+					navigator.returnToPocketCode(uri.getPath());
```

Comment:
We should consider using the uri here instead of a string path. I will create a ticket for this, so let us leave it for now though.

Reply:
:+1: